File: apps/apple_tv/Sources/AWAtv/Models/EpgProgramme.swift
import Foundation

/// One scheduled programme on a channel — XMLTV `<programme>`.
struct EpgProgramme: Codable, Hashable, Sendable {
    let channelTvgId: String
    let start: Date
    let stop: Date
    let title: String
    let description: String?
    let category: String?

    init(
        channelTvgId: String,
        start: Date,
        stop: Date,
        title: String,
        description: String? = nil,
        category: String? = nil
    ) {
        self.channelTvgId = channelTvgId
        self.start = start
        self.stop = stop
        self.title = title
        self.description = description
        self.category = category
    }

    enum CodingKeys: String, CodingKey {
        case channelTvgId = "channel_tvg_id"
        case start
        case stop
        case title
        case description
        case category
    }
}

File: packages/awatv_core/lib/src/models/epg_programme.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'epg_programme.freezed.dart';
part 'epg_programme.g.dart';

/// One scheduled programme on a channel (XMLTV `<programme>`).
@freezed
class EpgProgramme with _$EpgProgramme {
  const factory EpgProgramme({
    required String channelTvgId,
    required DateTime start,
    required DateTime stop,
    required String title,
    String? description,
    String? category,
  }) = _EpgProgramme;

  factory EpgProgramme.fromJson(Map<String, dynamic> json) =>
      _$EpgProgrammeFromJson(json);
}
